fleetshard/pkg/central/operator/upgrade_test.go

```diff
@@ -88,18 +109,108 @@ func TestOperatorUpgradeFreshInstall(t *testing.T) {
-	templateSpec := operatorDeployment.Object["spec"].(map[string]interface{})["template"].(map[string]interface{})["spec"]
-	assert.NotEmpty(t, templateSpec)
-	assert.Contains(t, templateSpec, "containers")
-	containers := templateSpec.(map[string]interface{})["containers"].([]interface{})
+	containers := operatorDeployment1.Spec.Template.Spec.Containers
 	assert.Len(t, containers, 2)
-	managerContainer := containers[1].(map[string]interface{})
-	assert.Equal(t, managerContainer["image"], operatorImage)
+	managerContainer := containers[1]
+	assert.Equal(t, managerContainer.Image, operatorImage1)
+}
+
+func TestOperatorUpgradeMultipleVersions(t *testing.T) {
+	fakeClient := testutils.NewFakeClientBuilder(t).Build()
+	u := NewACSOperatorManager(fakeClient)
+
+	operatorImages := []string{operatorImage1, operatorImage2}
+	err := u.InstallOrUpgrade(context.Background(), operatorImages)
+	require.NoError(t, err)
+
+	err = fakeClient.Get(context.Background(), client.ObjectKey{Namespace: operatorNamespace, Name: operatorDeployment1.Name}, operatorDeployment1)
+	require.NoError(t, err)
+	managerContainer := operatorDeployment1.Spec.Template.Spec.Containers[1]
+	assert.Equal(t, managerContainer.Image, operatorImage1)
+
+	err = fakeClient.Get(context.Background(), client.ObjectKey{Namespace: operatorNamespace, Name: operatorDeployment2.Name}, operatorDeployment2)
+	require.NoError(t, err)
+	managerContainer = operatorDeployment2.Spec.Template.Spec.Containers[1]
+	assert.Equal(t, managerContainer.Image, operatorImage2)
+}
+
+func TestOperatorUpgradeDoNotInstallLongTagVersion(t *testing.T) {
+	fakeClient := testutils.NewFakeClientBuilder(t).Build()
+	u := NewACSOperatorManager(fakeClient)
+
+	operatorImageWithLongTag := "quay.io/rhacs-eng/stackrox-operator:3.74.1-with-ridiculously-long-tag-version-name"
+	err := u.InstallOrUpgrade(context.Background(), []string{operatorImageWithLongTag})
+	require.Errorf(t, err, "zero tags parsed from images")
 
+	deployments := &appsv1.DeploymentList{}
+	err = fakeClient.List(context.Background(), deployments)
+	require.NoError(t, err)
+	assert.Len(t, deployments.Items, 0)
+}
+
+func TestParseOperatorImages(t *testing.T) {
+	cases := map[string]struct {
+		images     []string
+		expected   []map[string]string
+		shouldFail bool
+	}{
+		"should parse one valid operator image": {
+			images: []string{operatorImage1},
+			expected: []map[string]string{
+				{"repository": operatorRepository, "tag": "3.74.1"},
+			},
+		},
+		"should parse two valid operator images": {
+			images: []string{operatorImage1, operatorImage2},
+			expected: []map[string]string{
+				{"repository": operatorRepository, "tag": "3.74.1"},
+				{"repository": operatorRepository, "tag": "3.74.2"},
+			},
+		},
+		"should ignore duplicate operator images": {
+			images: []string{operatorImage1, operatorImage1},
+			expected: []map[string]string{
+				{"repository": operatorRepository, "tag": "3.74.1"},
+			},
+		},
+		"fail if image does contain colon": {
```

Comment:
Should this be supported ? 

```suggestion
		"should accept images from multiple repositories with the same tag": {
			images: []string{"repo1:tag","repo1:tag"},
			expected: []map[string]string{
				{"repository":"repo1", "tag": "tag"},
				{"repository":"repo2", "tag": "tag"},
			},
		},
		"fail if image does contain colon": {
```